packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -938,10 +939,16 @@ abstract class ResidentHandlers {
   /// If the device has a connected vmservice, this method will attempt to hide
   /// and restore the debug banner before taking the screenshot.
   ///
-  /// Throws an [AssertionError] if [Device.supportsScreenshot] is not true.
+  /// If the device type does not support a "native" screenshot, then this
+  /// will fallback to a rasterizer screenshot from the engine. This has the
+  /// downside of being unable to display the contents of platform views.
+  ///
+  /// Currently the behavior of screenshot is to bail if it encounters any
```

Comment:
What does "bail" mean here? Throw a tool exit? Silently fail? Fail with a printError? etc.

Maybe also rephrase without "Currently", like "Fails with an error message and continues if there is an RPC error".

Reply:
Took another pass on this